src/combinatorics/young_diagram.lean

```diff
@@ -123,4 +128,128 @@ end distrib_lattice
 /-- Cardinality of a Young diagram -/
 @[reducible] protected def card (μ : young_diagram) : ℕ := μ.cells.card
 
+section transpose
+/-- The `transpose` of a Young diagram is obtained by swapping i's with j's. --/
+
+def transpose (μ : young_diagram) : young_diagram :=
+{ cells :=  (equiv.prod_comm _ _).finset_congr μ.cells,
+  is_lower_set := λ _ _ h, by {
+    simp only [finset.mem_coe, equiv.finset_congr_apply, finset.mem_map_equiv],
+    intro hcell, apply μ.is_lower_set _ hcell, simp [h], } }
+
+@[simp] lemma mem_transpose {μ : young_diagram} {c : ℕ × ℕ} : c ∈ μ.transpose ↔ c.swap ∈ μ :=
+by { change c ∈ μ.cells.map _ ↔ _, rw finset.mem_map_equiv, refl, }
+
+@[simp] lemma transpose_eq_iff {μ ν : young_diagram} : μ.transpose = ν ↔ μ = ν.transpose :=
```

Comment:
This seems like it shouldn't be a simp lemma -- why should simp normal form be that `transpose` is on the right vs the left? A better simp lemma would be `μ.transpose = ν.transpose ↔ μ = ν`.

If you keep the current lemma, I'd call it `transpose_eq_if_eq_transpose`, reserving `transpose_eq_iff` for the above simp lemma.

Reply:
How would one prove `μ.transpose = ν ↔ μ = ν.transpose` then? (Or even one direction of it?)

I guess I agree there's no reason to prefer one side.

Reply:
Okay, I edited this section.